src/internal/converters/ics/ics.go

```diff
@@ -429,19 +476,66 @@ func FromJSON(ctx context.Context, body []byte) (string, error) {
 		if inline {
 			cidv, err := attachment.GetBackingStore().Get("contentId")
 			if err != nil {
-				return "", clues.Wrap(err, "getting attachment content id")
+				return clues.Wrap(err, "getting attachment content id")
 			}
 
 			cid, err := str.AnyToString(cidv)
 			if err != nil {
-				return "", clues.Wrap(err, "getting attachment content id string")
+				return clues.Wrap(err, "getting attachment content id string")
 			}
 
 			props = append(props, keyValues("CID", cid))
 		}
 
-		event.AddAttachment(base64.StdEncoding.EncodeToString(content), props...)
+		iCalEvent.AddAttachment(base64.StdEncoding.EncodeToString(content), props...)
 	}
 
-	return cal.Serialize(), nil
+	cancelledDates, err := getCancelledDates(ctx, event)
+	if err != nil {
+		return clues.Wrap(err, "getting cancelled dates")
+	}
+
+	dateStrings := []string{}
+	for _, date := range cancelledDates {
+		dateStrings = append(dateStrings, date.Format(iCalDateFormat))
+	}
+
+	if len(dateStrings) > 0 {
+		iCalEvent.AddProperty(ics.ComponentPropertyExdate, strings.Join(dateStrings, ","))
+	}
+
+	return nil
+}
+
+func getCancelledDates(ctx context.Context, event models.Eventable) ([]time.Time, error) {
+	cancelldedOcurrances := event.GetAdditionalData()["cancelledOccurrences"]
+	if cancelldedOcurrances == nil {
+		return nil, nil
+	}
+
+	dates := []time.Time{}
+
+	for _, occ := range cancelldedOcurrances.([]any) {
+		instance, err := str.AnyToString(occ)
+		if err != nil {
+			return nil, clues.Wrap(err, "getting cancelled occurrence id")
+		}
+
+		splits := strings.Split(instance, ".")
+		if len(splits) < 2 {
+			return nil, clues.NewWC(ctx, "invalid cancelled occurrence id").With("id", instance)
+		}
+
+		startStr := splits[len(splits)-1]
```

Comment:
is there similar code for this in the restore path for events? If there is, could we pull that out into a helper function so we don't have to keep two instances of it the same?